app/uk/gov/hmrc/hec/controllers/IFController.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hec.controllers
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.UUID
+
+import com.google.inject.{Inject, Singleton}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.hec.models.TaxYear
+import uk.gov.hmrc.hec.models.ids.{CTUTR, SAUTR}
+import uk.gov.hmrc.hec.services.{IFService, IFServiceImpl}
+import uk.gov.hmrc.hec.util.Logging
+import uk.gov.hmrc.hec.util.Logging.LoggerOps
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import cats.data.{NonEmptyList, Validated, ValidatedNel}
+import cats.data.Validated._
+import cats.implicits._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class IFController @Inject() (
+  IFService: IFService,
+  cc: ControllerComponents
+)(implicit ec: ExecutionContext)
+    extends BackendController(cc)
+    with Logging {
+
+  private def correlationId = UUID.randomUUID().toString
+
+  /**
+    * Fetch individual user's self-assessment status
+    * @param utr Self-assessment UTR
+    * @param taxYear Start tax year
+    * @return Self-assessment status with UTR and tax year
+    */
+  def getSAStatus(utr: String, taxYear: String): Action[AnyContent] = Action.async { implicit request =>
+    val sautrValidation   = SAUTR.fromString(utr).toValidNel("Invalid SAUTR")
+    val taxYearValidation = TaxYear.fromString(taxYear).toValidNel("Invalid tax year")
```

Comment:
Should we do validation on correlationId as well to check if it's a valid UUID? @AndyHWChung 
As this is already covered in stubs i was not sure if that check should be here or not

Reply:
the correlation id should be of type `UUID` and generated for each request (so it won't come from config) - atm it's being generated by the BE. The FE isn't passing it to the BE which I think is good enough - we'll be able to trace requests between the FE and BE if we need to.

Reply:
This has been enhanced in later commits @priyankadhiman3006 (sorry!) where as @AndyHWChung suggested, I've just passed down the correlationId as a `UUID` instead of `String`